tests/ctst/steps/put-object.ts

```diff
@@ -0,0 +1,86 @@
+import {Given, Then} from '@cucumber/cucumber';
+import Zenko from 'world/Zenko';
+import {S3, Utils} from 'cli-testing';
+import {ListObjectsV2Output, ListObjectVersionsOutput, PutObjectOutput} from '@aws-sdk/client-s3';
+import {safeJsonParse} from 'common/utils';
+import assert from 'assert';
+import {saveAsFile} from './utils/utils';
+
+Given('{int} threads each uploading {int} versions of object {string} of size {int} bytes', async function (
+    this: Zenko,
+    numberOfThread: number,
+    numberOfVerionPerThread: number,
+    objectName: string,
+    sizeBytes: number,
+) {
+    const bucketName = this.getSaved<string>('bucketName');
+    this.addToSaved('objectName', objectName);
+    let processedCounter = numberOfVerionPerThread * numberOfThread;
+
+    await Promise.all(Array.from({length: numberOfThread}, async () => {
+        for (let i = 0; i < numberOfVerionPerThread; i++) {
+            const tempFileName = `${Utils.randomString()}_${objectName}`;
+            const objectBody = 'a'.repeat(sizeBytes);
+            await saveAsFile(tempFileName, objectBody);
+
+            const result = await S3.putObject({
```

Comment:
To maximize our chance of hitting the race condition, maybe we should create the files first, then call the putObject in parallel, with a limit (async.parallelLimit?), as here we are doing the calls sequentially, so even with the 100 threads, we might be calling s3 almost never at the same times?

Another way to implement it would be to:
- First pre-create all files (or just use the same one for all as an optim?)
- Batch 100 calls to s3
- Wait for it to complete
- Do it 10 times

That way each batch will have much more chance to trigger the race condition, what do you think?

Reply:
Just a note on this, the bug is purely on the metadata layer so we could simplify the tests and make them run faster by uploading empty objects with no data.

Reply:
@williamlardier as checked together, I tried to rerun test suite to make sure we don't have more flaky tests. The CI seems to run with, from time to time, some flaky tests (never the new one). Seems that the current test suite is not degraded. So I guess we can move on.

@Kerkesni good point. I can update the code but I'm not sure it's a good move. We are in e2e test suite, not really a unit one or integration one. So we should try to be very close from production. That's why I would like to keep the file (even if it's not tested). Let me know what do you think, I don't have a strong opinion right now on it, as I'm still a new joiner.

Reply:
I think we could at least reuse the same file, or create them outside of this for loop?

We also have the `uploadSetup` and `uploadTeardown` that coul dbe helpful here, as they handle removing the resources after the call, to avoid filling the test platform.